typing/scalar.mli

```diff
@@ -0,0 +1,359 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                 Jacob Van Buren, Jane Street, New York                 *)
+(*                                                                        *)
+(*   Copyright 2025 Jane Street Group LLC                                 *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+(** This module defines the scalar types intrinsic to the OCaml compiler, and the
+    primitive operations defined on them.
+
+    A [Scalar.t] represents a particular OCaml type that represents a scalar value. It
+    might be tagged, boxed, or neither. There is also a type parameter for the locality of
+    the scalar value, which represents the location in which that boxed values are
+    allocated.
+
+    The important consideration is for a [Scalar.t] to represent all of the primitives
+    that we want to expose. The submodules are organized to make it easy for use different
+    subsets of scalars in different places. Some examples:
+
+    - Primitive arguments don't depend on the locality of their arguments, but the results
+    do.
+    - Some primitives only take integers, some take only floats, and Three_way_compare
+    takes any scalar type.
+    - The bytecode compiler wants to easily map unboxed/untagged values to their [value]
+    equivalents
+    - The middle-end wants to easily cast between any integraal values using only certain
+    primitives.
+*)
+
+type any_locality_mode = Any_locality_mode
+
+module Integer_comparison : sig
```

Comment:
Encountering this module immediately after reading the above comment was surprising - the comment says `scalar.ml` is about representing types, but the presence of `Integer_comparison` suggests it's also about representing operations on those types. Which I thought was the job of `primitive.ml`.

I guess this file has the representations of the "intrinsics" and primitive has the other primitives?  My instinct is the comment above should explain that distinction, or this stuff go over to `primitive.ml` (or both).